src/main/resources/hudson/plugins/ec2/EC2Cloud/computerSet.jelly

```diff
@@ -22,28 +22,32 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 -->
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler"
+         xmlns:l="/lib/layout" xmlns:dd="/lib/layout/dropdowns" xmlns:f="/lib/form">
   <j:if test="${it.hasPermission(it.PROVISION)}">
+    <j:set var="formId" value="${h.generateId()}"/>
     <tr>
       <td />
-      <td colspan="${monitors.size()+1}">
-        <f:form action="${rootURL}/${it.url}/provision" method="post" name="provision">
-          <input type="submit" class="ec2-provision-button" value="${%Provision via} ${it.displayName}" />
-          <select name="template">
-            <j:forEach var="t" items="${it.templates}">
-              <option value="${t.description}">${t.displayName}</option>
-            </j:forEach>
-          </select>
-          <st:once>
-            <script>
-              Behaviour.register({
-                ".ec2-provision-button" : function (e) {
-                  new YAHOO.widget.Button(e, { type: "menu", menu: e.nextSibling });
-                }
-              });
-            </script>
-          </st:once>
+      <td colspan="${monitors.size()+2}">
+        <l:overflowButton icon="symbol-add"
+                          text="${%Provision via} ${it.displayName}"
+                          tooltip="${null}"
+                          clazz="jenkins-!-margin-top-2">
+          <j:forEach var="t" items="${it.templates}">
+            <dd:custom>
+              <button class="jenkins-dropdown__item"
+                      data-type="ec2-provision"
+                      data-form="${formId}"
+                      data-url="${t.description}">
+                ${t.displayName}
+              </button>
+            </dd:custom>
+          </j:forEach>
+        </l:overflowButton>
+        <f:form action="${rootURL}/${it.url}/provision" method="post" name="provision" id="${formId}">
```

Comment:
you could probably move it so the form is a parent of the j:forEach  (like it was before roughly)  and then use something like`e.closest('form')` to save you having to generate Ids.

(non-blocking)

Reply:
I think that doesn't work because of how `overflowButton` works: the elements in `foreach` are added to `<template>` which is the cloned into a popup that's no longer part of the `form` but floats above it.

It should be possible to remove the form and use `fetch` like the OpenStack plugin does (https://github.com/jenkinsci/openstack-cloud-plugin/blob/fc5384db4d5f1f4f2ebdec2aec7e28ce312cf0df/plugin/src/main/resources/jenkins/plugins/openstack/compute/JCloudsCloud/computerSet.jelly#L23), but I didn't want to change the behavior.